### PR TITLE
♫ add playback bar

### PR DESCRIPTION
**Pull request by anuejn**
♫🔊🎤🎶

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -73,7 +73,7 @@ repos:
     hooks:
       - id: format
         name: format
-        entry: pnpm --prefix frontend/ run check:format
+        entry: pnpm --prefix frontend/ run format
```
Comment (pajowu) [inline .pre-commit-config.yaml:76]: 😍

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -1,24 +0,0 @@
-import { Route, Router, Switch } from 'wouter';
```
Comment (pajowu) [inline frontend/src/App.tsx:1]: Why do we lowercase all filesnames?

File: frontend/src/components/popup.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      {show ? (
+        <div
+          className={clsx(
+            'p-4',
+            'bg-white',
+            'border-black',
+            'border-2',
+            'shadow-brutal',
+            'rounded-lg',
+            'm-2',
```
Comment (pajowu) [inline frontend/src/components/popup.tsx:41]: Popper does not like margins on itself (I think this line is was causes it). The popper element reports in the console:

> CSS "margin" styles cannot be used to apply padding between the popper and its reference element or boundary. To replicate margin, use the `offset` modifier, as well as the `padding` option in the `preventOverflow` and `flip` modifiers. 

More Details: https://popper.js.org/docs/v2/migration-guide/#4-remove-all-css-margins

File: frontend/src/pages/new_document.tsx [hunk 4]
```diff
@@ -48,11 +47,7 @@ export default function NewDocumentPage() {
   };
 
   return (
-    <div
-      className="h-screen p-6 flex items-center justify-center"
-      onDragOver={(e) => e.preventDefault()}
-      onDrop={(e) => e.preventDefault()}
-    >
+    <AppCenter onDragOver={(e) => e.preventDefault()} onDrop={(e) => e.preventDefault()}>
       <Dialog>
```
Comment (pajowu) [inline frontend/src/pages/new_document.tsx:51]: Can I have a back button for the new document dialog too?

File: frontend/src/pages/user_home.tsx [hunk 5]
```diff
@@ -52,7 +56,30 @@ export default function HomePage() {
...
+              'border',
+              'border-gray-200',
+              'hover:shadow-lg',
+              'hover:scale-105',
+              'transition-all',
+              'flex',
+            )}
+          >
+            {' '}
+            <AiOutlinePlus size="1.75rem" />
```
Comment (pajowu) [inline frontend/src/pages/user_home.tsx:79]: I think the aria label (or `visually-hidden` span) is missing here for accessibility.

File: frontend/src/common/top_bar.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import { IconButton, PrimaryButton } from '../components/button';
+import { Popup } from '../components/popup';
+
+export const TopBar = primitiveWithClassname('div', 'mb-8 flex items-center gap-4 justify-between');
+export const TopBarTitle = primitiveWithClassname('h2', 'text-xl font-bold');
+export const TopBarPart = primitiveWithClassname('div', 'gap-4 flex items-center');
+
+export function MeButton() {
+  return (
+    <Popup button={<IconButton icon={BiUser} />}>
```
Comment (pajowu) [inline frontend/src/common/top_bar.tsx:15]: This is missing an aria label / visually hidden span

**Review by pajowu — CHANGES_REQUESTED** [review]
Looks good in general, just some smaller changes (I think most of them are from #52)

File: frontend/src/pages/new_document.tsx (hunk 4 above)
Comment (phlmn) [reply frontend/src/pages/new_document.tsx:51]: Maybe a cancel button instead?

File: frontend/src/components/popup.tsx (hunk 3 above)
Comment (anuejn) [reply frontend/src/components/popup.tsx:41]: ah good catch!

File: frontend/src/App.tsx (hunk 2 above)
Comment (pajowu) [reply frontend/src/App.tsx:1]: Nope, was just wondering because I always found it intuitive to have the main export as the filename

**Comment by pajowu** [comment]
I noticed that the player bar waveform is not shown until the editor state is fully loaded. The cause seems to be that both editor state processing and waveform generation are done on the main thread. It might be a good idea to move this to a webworker or similar (see also https://github.com/wavesurfer-js/wavesurfer.js/issues/1867 ). But that doesn't have to happen in this PR

**Review by pajowu — APPROVED** [review]

**Comment by anuejn** [comment]
Sounds good :). I do think though, that we want both the wavesurfer loading as well as the automerge merging to happen in a webworker. But lets do that in the future :)

**Pull request merged**